### PR TITLE
Add meme extension

### PR DESCRIPTION
**Pull request by mattdrose**
## Description

This PR adds a new extension for generating memes.

It uses the [imgflip API](https://imgflip.com/api). You can search through meme templates, and then input text to generate a custom meme hosted on imgflip.

## Type of change

- New extension

## Screencast

https://user-images.githubusercontent.com/1107972/151650132-95174c85-3406-4b69-b667-ffa37f35ed4e.mp4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

File: extensions/meme/src/api/api.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
+import fetch from 'node-fetch';
+import {ImgflipCaptionImageResponse, ImgflipGetMemesResponse} from './types';
+import {Meme} from 'types';
+import {getLocalStorageItem, setLocalStorageItem} from '@raycast/api';
+
+const IMGFLIP_USERNAME = 'raycast';
+const IMGFLIP_PASSWORD = 'aha_qkt1wnh2QGJ-hrv';
```
Comment (mattdrose) [inline extensions/meme/src/api/api.ts:7]: Not sure if there's a way to encrypt credentials. The same credentials are used to login to the imgflip account and authenticate API requests. I created account this specifically for raycast.

File: extensions/meme/package.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  "name": "meme",
+  "title": "Meme Generator",
+  "description": "Generate memes using the Imgflip API",
+  "icon": "command-icon.png",
+  "author": "mattrose",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "meme",
+      "title": "Generate meme",
```
Comment (pernielsentikaer) [inline extensions/meme/package.json:12]:
```suggestion
      "title": "Generate meme",
      "subtitle": "imgflip",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @mattdrose 💪 

Thanks for your contribution, you should add credentials as [preferences](https://developers.raycast.com/api-reference/preferences) and then validate if wrong credentials.

The test credentials are fine to have, then I can test with them 😄 

I'll do a review as soon as you have added preference 😏

**Comment by mattdrose** [comment]
I just heard back from imgur. They don't set an API limit, and won't unless usage is affecting their server performance. Credentials per user can be introduced if the extension ever becomes popular enough to affect their servers in this way.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @mattdrose 🔥 

What an awesome extension 💪 

Some feedback for you:

- Your extensions have [empty state flicker](https://developers.raycast.com/basics/prepare-an-extension-for-store#empty-states) - Could you address this?

- There is something with the placeholder, both can't be optional, do you have an idea on how to improve it? 

NkJTru7iTCVcXPaL

- Your ActionPanel items have icons, except `Generate` – you could add an icon here too

Looking forward to testing it again 💯 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by sxn** [comment]
Hi @mattdrose! 
As there hasn't been any activity here for two months, I'm going to close this PR now. Feel free to reopen it or open a new PR whenever you want it to be reviewed again! 😸

**Pull request closed without merging**